test/unit/widgets/forms.test.ts

```diff
@@ -1,1435 +1,1455 @@
+      )
+    ).toMatchInlineSnapshot(`
+     <div
+       style="padding-left: 2em; border: 0.05em solid #888888;"
+     />
+    `)
+  })
+  // @@ TODO check this further what test could I use to make sure
+  // to test that the form gets added but obviously not in the container passed in
+  it.skip('still returns an element if the container is null', () => {
+    // debugger
+    const container = document.createElement('div')
+    const already = {}
+    const subject = namedNode('http://example.com/#this')
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
+    expect(
+      field[ns.ui('Form').uri](
+        document,
+        container,
+        already,
+        subject,
+        form,
+        store,
+        callbackFunction
+      )
+    ).toMatchInlineSnapshot(`
+      <div>
+        <h3
+          style="[object Object]"
+        >
+          [object Object]
+        </h3>
+      </div>
+    `)
+  })
+  // @@ TODO need to double check proper subjects and what gets returned
+  it.skip('returns.. if the subject has already been processed...', () => {
+    const dubSubject = namedNode('subject')
+    const container = document.createElement('container')
+    const already = { dubSubject }
+    const subject = dubSubject
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
+    expect(
+      field[ns.ui('Form').uri](
+        document,
+        container,
+        already,
+        subject,
+        form,
+        store,
+        callbackFunction
+      )
+    ).toMatchInlineSnapshot(`
+      <div>
+        <h3
+          style="[object Object]"
+        >
+          [object Object]
+        </h3>
+      </div>
+    `)
+  })
+})
+
+describe('Options field', () => {
+  it('exists', () => {
+    expect(field[ns.ui('Options').uri]).toBeInstanceOf(Object)
+  })
+  it('runs', () => {
+    const container = document.createElement('div')
+    const already = {}
+    const subject = namedNode('http://example.com/#this')
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
+    expect(
+      field[ns.ui('Options').uri](
+        document,
+        container,
+        already,
+        subject,
+        form,
+        store,
+        callbackFunction
+      )
+    ).toMatchInlineSnapshot('<div />')
+  })
+})
+
+describe('Multiple field', () => {
+  it('exists', () => {
+    expect(field[ns.ui('Multiple').uri]).toBeInstanceOf(Object)
+  })
+  it.skip('runs', () => {
+    const container = document.createElement('div')
+    const already = {}
+    const subject = namedNode('http://example.com/#this')
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
```

Comment:
I see a lot of these in the test file, but seldom actually a test checking when it's run. Please write tests that check when they are called.

Reply:
This PR only re-enables the tests that you deleted when you switched rdflib from a mock to a spy.
The existing tests only test how form fields are created, not how you can interact with them, that's why none of those callbacks actually get called in any of the tests.

I created a follow-up issue https://github.com/solid/solid-ui/issues/263 so that we can merge this and add the interaction tests in additional set of tests.